src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs

```diff
@@ -135,45 +136,40 @@ public void InitializeService(ServiceHost serviceHost)
                     {
                         if (MetadataScriptTempFileStream.IsScriptTempFileValid(connectionInfo.ConnectionDetails.ServerName))
                         {
-                            await requestContext.SendResult(true);
+                            return;
                         }
 
-                        var scripts = SmoScripterFactory.GenerateAllServerTableScripts(sqlConn);
+                        var scripts = await SmoScripterFactory.GenerateAllServerTableScripts(sqlConn);
                         if (scripts != null)
                         {
                             try
                             {
                                 MetadataScriptTempFileStream.Write(connectionInfo.ConnectionDetails.ServerName, scripts);
-                                await requestContext.SendResult(true);
                             }
                             catch (Exception ex)
                             {
                                 Logger.Error($"An error was encountered while writing to the cache. Error: {ex.Message}");
-                                await requestContext.SendError(ex);
```

Comment:
sending these errors back isn't needed anymore?

Reply:
There currently is no mechanism to return errors from event handlers.

Reply:
So if you're expecting to have errors and handle them in some way on the client side you should be using a request. Notifications are for things that don't expect a response of any sort back. Does your design have any plans for how to handle issues happening when initializing the metadata?

Reply:
I'm not expecting any major errors. In the worst case, if STS is not able to write scripts to the temp directory, then we wouldn't be able to supply Copilot with any context and Copilot would work as it currently does. To mitigate that, the generated scripts could then be held in memory, but then we would probably run into high memory consumption issues depending on how many connections a user has and the size of those connections.

Reply:
Sounds reasonable to me, you can always change it later if needed. Keeping it simpler for now seems like a good idea.